packages/web/src/components/molecules/DeleteEntityModal/DeleteEntityModal.tsx

```diff
@@ -2,101 +2,57 @@ import {useState} from 'react';
 }> = props => {
-  const {onCancel, useDeleteMutation, name, onConfirm, entityLabel, defaultStackRoute, idToDelete} = props;
-
-  const {close} = useModal();
-  const back = useDashboardNavigate(defaultStackRoute);
-
-  const onEvent = usePressEnter();
-
-  const [deleteEntity] = useDeleteMutation();
+  const {onDelete, name, onClose, onConfirm, entityLabel, defaultStackRoute, idToDelete} = props;
 
   const [checkName, setName] = useState('');
 
-  const onDelete = () => {
-    deleteEntity(idToDelete || name)
-      .then(displayDefaultNotificationFlow)
-      .then(() => {
-        notificationCall('passed', `${capitalize(entityLabel)} was successfully deleted.`);
-        close();
-
-        if (onConfirm) {
-          onConfirm();
-        } else {
-          back();
-        }
-      })
-      .catch(error => {
-        notificationCall('failed', error.title, error.message);
-      });
-  };
-
   return (
-    <FullWidthSpace
-      size={24}
-      direction="vertical"
-      onKeyPress={event => {
-        if (name === checkName) {
-          onEvent(event, onDelete);
-        }
-      }}
-    >
-      <Text className="regular middle" color={Colors.slate400}>
-        Do you really want to delete this {entityLabel}?
-        <br />
-        All your historical and analytical data will also be removed.
-      </Text>
-      <Text className="regular middle" color={Colors.slate400}>
-        Please enter the name of this {entityLabel} (<span style={{color: Colors.whitePure}}>{name}</span>) to delete it
-        forever.
-      </Text>
-      <Input
-        placeholder={`${capitalize(entityLabel)} name`}
-        onChange={e => {
-          setName(e.target.value);
-        }}
-      />
-      <FooterSpace size={10}>
-        <Button $customType="secondary" onClick={onCancel}>
-          Cancel
-        </Button>
-        <Button $customType="warning" disabled={name !== checkName} onClick={onDelete}>
-          Delete
-        </Button>
-      </FooterSpace>
-    </FullWidthSpace>
+    <DeleteModal
+      onDelete={onDelete}
+      onClose={onClose}
+      onConfirm={onConfirm}
+      successMessage={`${capitalize(entityLabel)} was successfully deleted.`}
+      defaultStackRoute={defaultStackRoute}
+      idToDelete={idToDelete || name}
+      actionDisabled={checkName !== name}
+      deleteOnEnter
+      content={
+        <FullWidthSpace size={24} direction="vertical">
+          <Text className="regular middle" color={Colors.slate400}>
+            Do you really want to delete this {entityLabel}?
+            <br />
+            All your historical and analytical data will also be removed.
+          </Text>
+          <Text className="regular middle" color={Colors.slate400}>
+            Please enter the name of this {entityLabel} (<span style={{color: Colors.whitePure}}>{name}</span>) to
+            delete it forever.
+          </Text>
+          <Input
+            placeholder={`${capitalize(entityLabel)} name`}
```

Comment:
Not important: Could it be disabled during request?

Reply:
loading state is handled in the child component `DeleteModal`. consider not to do it, because we would have to move that state one level above and handle it for each use case

Reply:
It doesn't have to be for each specific use case, it can be based on `onDelete` async result

Reply:
exactly. that's what we're currently doing inside of `DeleteModal`. what I meant was, if we move this implementation from there, we would have to duplicate it in all the places were using the component.